packages/react-router/modules/Route.js

```diff
@@ -62,15 +63,15 @@ class Route extends React.Component {
     )
 
     this.setState({
-      match: this.computeMatch(nextProps)
+      match: this.computeMatch(nextProps, nextContext)
```

Comment:
Need to use the next context object here. Otherwise when there is no `location` prop, the previous location will be used when computing the new match.